src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -241,25 +239,20 @@ class GcsRpcClient {
       std::pair<std::string, int> address;
       int index = 0;
       for (; index < RayConfig::instance().ping_gcs_rpc_server_max_retries(); ++index) {
-        address = get_server_address_();
-        RAY_LOG(DEBUG) << "Attempt to reconnect to GCS server: " << address.first << ":"
-                       << address.second;
-        if (Ping(address.first, address.second, 100)) {
-          RAY_LOG(INFO) << "Reconnected to GCS server: " << address.first << ":"
-                        << address.second;
-          break;
+        if (get_server_address_(&address)) {
```

Comment:
Better also move this (recreating the RPC client) to also ServiceBasedGcsClient. So we can handle the RPC stuff and the pub sub stuff in one place.

Reply:
fixed

Reply:
fixed